kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java

```diff
@@ -36,9 +39,21 @@
   
   void setMetadata(ObjectMeta metadata);
   
-  String getKind();
+  default String getKind() {
+    final Class<? extends HasMetadata> clazz = getClass();
+    final Kind kind = clazz.getAnnotation(Kind.class);
+    return kind != null ? kind.value() : clazz.getSimpleName();
+  }
   
-  String getApiVersion();
+  default String getApiVersion() {
+    final Class<? extends HasMetadata> clazz = getClass();
+    final ApiGroup group = clazz.getAnnotation(ApiGroup.class);
+    if (group != null) {
+      final ApiVersion version = clazz.getAnnotation(ApiVersion.class);
+      return group.value() + "/" + version.value();
```

Comment:
I think `version.value()` might produce `NullPointerEception` here in case only ApiGroup annotation is provided.

Reply:
Good catch!